aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -0,0 +1,666 @@
+   *
+   * <p>Example: iceberg.namespace=namespaceName
+   */
+  public static final String S3_TAG_ICEBERG_NAMESPACE = "iceberg.namespace";
+
+  /**
+   * Used by {@link S3FileIO} to tag objects when deleting. When this config is set, objects are
+   * tagged with the configured key-value pairs before deletion. This is considered a soft-delete,
+   * because users are able to configure tag-based object lifecycle policy at bucket level to
+   * transition objects to different tiers.
+   *
+   * <p>For more details, see
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html
+   *
+   * <p>Example: s3.delete.tags.my_key=my_val
+   */
+  public static final String S3_DELETE_TAGS_PREFIX = "s3.delete.tags.";
+
+  /**
+   * Number of threads to use for adding delete tags to S3 objects, default to {@link
+   * Runtime#availableProcessors()}
+   */
+  public static final String S3FILEIO_DELETE_THREADS = "s3.delete.num-threads";
+
+  /**
+   * Determines if {@link S3FileIO} deletes the object when io.delete() is called, default to true.
+   * Once disabled, users are expected to set tags through {@link #S3_DELETE_TAGS_PREFIX} and manage
+   * deleted files through S3 lifecycle policy.
+   */
+  public static final String S3_DELETE_ENABLED = "s3.delete-enabled";
+
+  public static final boolean S3_DELETE_ENABLED_DEFAULT = true;
+
+  /**
+   * Determines if S3 client will use the Acceleration Mode, default to false.
+   *
+   * <p>For more details, see
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html
+   */
+  public static final String S3_ACCELERATION_ENABLED = "s3.acceleration-enabled";
+
+  public static final boolean S3_ACCELERATION_ENABLED_DEFAULT = false;
+
+  /**
+   * Determines if S3 client will use the Dualstack Mode, default to false.
+   *
+   * <p>For more details, see
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/dual-stack-endpoints.html
+   */
+  public static final String S3_DUALSTACK_ENABLED = "s3.dualstack-enabled";
+
+  public static final boolean S3_DUALSTACK_ENABLED_DEFAULT = false;
+
+  /**
+   * Used by {@link S3FileIO}, prefix used for bucket access point configuration. To set, we can
+   * pass a catalog property.
+   *
+   * <p>For more details, see https://aws.amazon.com/s3/features/access-points/
+   *
+   * <p>Example: s3.access-points.my-bucket=access-point
+   */
+  public static final String S3_ACCESS_POINTS_PREFIX = "s3.access-points.";
+
+  /**
+   * This flag controls whether the S3 client will be initialized during the S3FileIO
+   * initialization, instead of default lazy initialization upon use. This is needed for cases that
+   * the credentials to use might change and needs to be preloaded.
+   */
+  public static final String S3_PRELOAD_CLIENT_ENABLED = "s3.preload-client-enabled";
+
+  public static final boolean S3_PRELOAD_CLIENT_ENABLED_DEFAULT = false;
+
+  private String s3FileIoSseType;
```

Comment:
can we take this chance to standardize the variable names? currently there is a mixture of `s3FileIoXXX` and `s3XXXX`, I think they should all match the config names and be `s3XXX`.